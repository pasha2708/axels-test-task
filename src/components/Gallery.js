import React from 'react';
import { Switch, Route, Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import Row from 'react-bootstrap/Row';

import { ModalWindow } from '../components';
import { StyledCol, StyledContainer } from '../styled/components/Gallery';

import { GET_PREVIEW } from '../redux/ducks/gallery';

const Gallery = ({ data }) => {
  const dispatch = useDispatch();
  React.useEffect(() => {
    dispatch({type: GET_PREVIEW});
  }, [dispatch]);

  return (
    <StyledContainer>
      <Row>
        {data.images &&
          data.images.map((item) => (
           
              <StyledCol key={item.id} md={6} lg={4}>
                <Link to={`/images/${item.id}`}>
                  <img src={item.url} alt="preview" />
                </Link>
                <Switch>
                <Route path={`/images/${item.id}`}>
                  <ModalWindow id={item.id} data={data} />
                </Route>
              </Switch>
              </StyledCol>
              
            
          ))}
      </Row>
    </StyledContainer>
  );
};

export default Gallery;
