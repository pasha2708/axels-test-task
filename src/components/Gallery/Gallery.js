import React from "react";
import axios from "axios";
import { Col, Container } from "react-bootstrap";
import "bootstrap/dist/css/bootstrap.min.css";
import Row from "react-bootstrap/Row";
import styled from "styled-components";

const Substrate = styled.div`
  display: block;
  text-align: center;
`;

function Gallery(props) {
  const [data, setData] = React.useState([]);

  React.useEffect(() => {
    axios
      .get("https://boiling-refuge-66454.herokuapp.com/images")
      .then((res) => {
        setData(res.data);
      });
  }, []);

  return (
    <div>
      <Container>
        <Substrate>
          <Row>
            {data.map((item) => (
              <Col key={item.id} style={{ marginBottom: "20px" }} md={6} lg={4}>
                <img onClick={props.setActive} src={item.url} alt="preview" />
              </Col>
            ))}
          </Row>
        </Substrate>
      </Container>
    </div>
  );
}

export default Gallery;
