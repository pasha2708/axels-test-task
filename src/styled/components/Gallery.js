import { Col, Container } from 'react-bootstrap';
import styled from 'styled-components';

const StyledCol = styled(Col)`
  * {
    margin-bottom: 1rem;
  }
`;

const StyledContainer = styled(Container)`
  display: block;
  text-align: center;
  img {
    cursor: pointer;
  }
`;

export { StyledCol, StyledContainer };