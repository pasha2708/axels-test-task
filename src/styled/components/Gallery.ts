import { Col, Container } from "react-bootstrap";
import styled from "styled-components";

const StyledCol = styled(Col)`
	img {
		margin-bottom: 1rem;
	}
`;

const StyledContainer = styled(Container)`
	text-align: center;
	img {
		cursor: pointer;
	}
`;

export { StyledCol, StyledContainer };
