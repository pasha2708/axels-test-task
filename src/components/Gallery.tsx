import React from 'react';
import { Routes, Route, Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

import { ModalWindow } from '../components';
import { StyledCol, StyledContainer } from '../styled/components/Gallery';
import { sagaActions } from '../redux/sagaActions';
import { BasicStateTypes } from '../redux/reducer';
import styled from 'styled-components';
import { CircularProgress } from '@mui/material';

const GalleryStyle = styled.div`
	display: flex;
	justify-content: center;
	flex-wrap: wrap;
`;

const Gallery: React.FC = () => {
	const images = useSelector((state: BasicStateTypes) => state.images);
	const dispatch = useDispatch();
	React.useEffect(() => {
		dispatch({ type: sagaActions.FETCH_IMAGES });
	}, [dispatch]);

	return (
		<StyledContainer>
			<GalleryStyle>
				{!images ? (
					<CircularProgress />
				) : (
					images.map((item) => (
						<StyledCol key={item.id}>
							<Link to={`/images/${item.id}`}>
								<img src={item.url} alt='preview' />
							</Link>
						</StyledCol>
					))
				)}
				<Routes>
					<Route path={'/images/:id'} element={<ModalWindow />} />
				</Routes>
			</GalleryStyle>
		</StyledContainer>
	);
};

export default Gallery;
