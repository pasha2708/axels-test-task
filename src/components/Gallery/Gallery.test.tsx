import { shallow, ShallowWrapper } from 'enzyme';
import { Link } from 'react-router-dom';

import Gallery from './Gallery';

const mockedImage = {
  id: 222,
  url: 'someURL/'
}

let component: ShallowWrapper;
const mockUseDispatch = jest.fn();
const mockUseLocation = jest.fn();
const setUp = () => shallow(<Gallery />);

beforeEach(async () => {
  component = await setUp();
});

afterEach(() => {
  jest.resetAllMocks();
});

jest.mock('react-redux', () => ({
  ...jest.requireActual<object>('react-redux'),
  useDispatch: () => mockUseDispatch,
  useSelector: () => ([mockedImage, mockedImage])
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual<object>('react-router-dom'),
  useLocation: () => mockUseLocation
}));

describe('GalleryPage component', () => {
  it('should take snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render all images from list', () => {
    expect(component.find(Link).children()).toHaveLength(2);
  });
});


