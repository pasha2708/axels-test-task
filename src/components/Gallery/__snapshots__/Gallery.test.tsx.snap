// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryPage component should take snapshot 1`] = `
<Styled(Container)>
  <Row />
</Styled(Container)>
`;
